src/vs/workbench/contrib/debug/browser/disassemblyView.ts

```diff
@@ -0,0 +1,209 @@
+import { editorBackground } from 'vs/platform/theme/common/colorRegistry';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { EditorPane } from 'vs/workbench/browser/parts/editor/editorPane';
+import { EditorInput } from 'vs/workbench/common/editor/editorInput';
+import { DISASSEMBLY_VIEW_ID } from 'vs/workbench/contrib/debug/common/debug';
+
+interface IDisassembledInstructionEntry {
+	allowBreakpoint: boolean;
+	isBreakpointSet: boolean;
+	instruction: DebugProtocol.DisassembledInstruction;
+}
+
+export class DisassemblyView extends EditorPane {
+
+	private static readonly NUM_INSTRUCTIONS_TO_LOAD = 50;
+
+	private _fontInfo: BareFontInfo;
+	private _disassembledInstructions: WorkbenchTable<IDisassembledInstructionEntry> | null;
+
+
+	constructor(
+		@ITelemetryService telemetryService: ITelemetryService,
+		@IThemeService themeService: IThemeService,
+		@IStorageService storageService: IStorageService,
+		@IConfigurationService configurationService: IConfigurationService,
+		@IInstantiationService private readonly _instantiationService: IInstantiationService
+	) {
+		super(DISASSEMBLY_VIEW_ID, telemetryService, themeService, storageService);
+
+		this._fontInfo = BareFontInfo.createFromRawSettings(configurationService.getValue('editor'), getZoomLevel(), getPixelRatio());
+		configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration('editor')) {
+				this._fontInfo = BareFontInfo.createFromRawSettings(configurationService.getValue('editor'), getZoomLevel(), getPixelRatio());
+				this._disassembledInstructions?.rerender();
+			}
+		});
+
+		this._disassembledInstructions = null;
+	}
+
+	protected createEditor(parent: HTMLElement): void {
+		const lineHeight = this._fontInfo.lineHeight;
+		const delegate = new class implements ITableVirtualDelegate<IDisassembledInstructionEntry>{
+			headerRowHeight: number = 0; // No header
+			getHeight(row: IDisassembledInstructionEntry): number {
+				return lineHeight;
+			}
+		};
+
+		this._disassembledInstructions = this._register(this._instantiationService.createInstance(WorkbenchTable,
+			'DisassemblyView', parent, delegate,
+			[
+				{
+					label: '',
+					tooltip: '',
+					weight: 0,
+					minimumWidth: 40,
+					maximumWidth: 40,
+					templateId: BreakpointRenderer.TEMPLATE_ID,
+					project(row: IDisassembledInstructionEntry): IDisassembledInstructionEntry { return row; }
+				},
+				{
+					label: 'instructions',
+					tooltip: '',
+					weight: 0.3,
+					templateId: InstructionRenderer.TEMPLATE_ID,
+					project(row: IDisassembledInstructionEntry): IDisassembledInstructionEntry { return row; }
+				},
+			],
+			[
+				this._instantiationService.createInstance(BreakpointRenderer),
+				this._instantiationService.createInstance(InstructionRenderer),
+			],
+			{
+				identityProvider: { getId: (e: IDisassembledInstructionEntry) => e.instruction.address },
+				horizontalScrolling: false,
```

Comment:
Make sure to also have an `accessibilityProvider`

Reply:
Will add a TODO to make sure I don't forget this. I wanted to add it after adding some proper rendering.

Reply:
Great thanks. As for inspiration you can checkout some other `accessibilityProviders` in debug lang. Try to have concise but descriptive messages with the least important piece of info at the end.